pkg/controllermanager/controller/shoot/shoot_control.go

```diff
@@ -87,102 +83,54 @@ func (c *Controller) shootDelete(obj interface{}) {
 	c.getShootQueue(obj).Add(key)
 }
 
-func (c *Controller) reconcileShootKey(key string) error {
-	ctx := context.TODO()
-	namespace, name, err := cache.SplitMetaNamespaceKey(key)
-	if err != nil {
-		return err
-	}
-
-	shoot, err := c.shootLister.Shoots(namespace).Get(name)
-	if apierrors.IsNotFound(err) {
-		logger.Logger.Debugf("[SHOOT RECONCILE] %s - skipping because Shoot has been deleted", key)
-		return nil
-	}
-	if err != nil {
-		logger.Logger.Infof("[SHOOT RECONCILE] %s - unable to retrieve object from store: %v", key, err)
-		return err
-	}
-
-	var (
-		shootLogger  = logger.NewShootLogger(logger.Logger, shoot.ObjectMeta.Name, shoot.ObjectMeta.Namespace, "")
-		needsRequeue = true
-		reconcileErr error
-	)
-
-	// Ignore Shoots which do not have the gardener finalizer.
-	if shoot.DeletionTimestamp != nil && !sets.NewString(shoot.Finalizers...).Has(gardenv1beta1.GardenerName) {
-		shootLogger.Debug("Do not need to do anything as the Shoot does not have my finalizer")
-		c.getShootQueue(shoot).Forget(key)
-		return nil
-	}
+func (c *Controller) respectSyncPeriodOverwrite() bool {
+	respectOverwrite := c.config.Controllers.Shoot.RespectSyncPeriodOverwrite
+	return respectOverwrite != nil && *respectOverwrite
+}
 
+func (c *Controller) checkShootIsOnHealthySeed(shoot *gardenv1beta1.Shoot) error {
 	seedName := shoot.Spec.Cloud.Seed
 	if seedName == nil {
```

Comment:
I think this can be removed as this function is only called when there is a seed for this shoot

Reply:
This function is called two times - once in the `deleteShoot` and once in `reconcileShoot` - Also, I guess, this makes readability of the overall flow easier. Did you mean sth different?

Reply:
Nvm, I found out what you meant.